lessons/4.mdx

```diff
@@ -0,0 +1,818 @@
+---
+title: Testing your Tier NFT
+description:
+  Prove your Tier NFT contract works with automated tests! Builds on knowledge
+  from Lesson 3.
+icons: ['solidity', 'hardhat', 'openzeppelin']
+---
+
+# Lesson 4: Testing your Tier NFT
+
+In this lesson, we'll be adding automated tests for our smart contract created
+in Lesson 3.
+
+Here you'll:
+
+- Learn about the benefits of writing automated tests, general testing concepts
+  and best practices
+- Walk away with many practical tests for smart contracts
+- Nibble on some food for thought about test-driven development
+
+## Why automated testing?
+
+Testing can:
+
+- prove code actually works
+- help you write more modular code
+- ensure code doesn't break when you or others make changes or add new features.
+  People new to the code-base may not even know when something breaks without
```

Comment:
@briangershon it's possible you'll see a few of these, but I'm going to suggest putting *new/unknown* language in italics. Just so the reader feels less of a plonker hehe
*code-base* is the first one that should be in italics! haha

Reply:
Will word it without lingo